server/handler/issue_comment.go

```diff
@@ -141,9 +137,18 @@ func (h *IssueComment) detectAndLogTampering(ctx context.Context, prctx pull.Con
 	return true
 }
 
-func (h *IssueComment) affectsApproval(actualComment string, rules []*approval.Rule) bool {
+func (h *IssueComment) affectsApproval(event github.IssueCommentEvent, rules []*approval.Rule) bool {
+	var originalBody string
+	switch event.GetAction() {
+	case "edited":
+		originalBody = event.GetChanges().GetBody().GetFrom()
+
+	case "created", "deleted":
+		originalBody = event.GetComment().GetBody()
+	}
+
 	for _, rule := range rules {
-		if rule.Options.GetMethods().CommentMatches(actualComment) {
+		if rule.Options.GetMethods().CommentMatches(originalBody) {
```

Comment:
how about this? this will look at all events but only run them through evaluation if the current or previous matches a rule pattern.

Reply:
oh right, i see what you mean now.